From c45f5e81ba75f1a1d41f49d4c6560234b7d1d6c7 Mon Sep 17 00:00:00 2001
From: Youming Ye <paulye@google.com>
Date: Mon, 4 Mar 2019 11:00:40 -0800
Subject: [PATCH 3/3] Change UI to be consistency with dialer settings light M2
 theme

Style changes: Change font categories based on dialer settings.
Change Toolbar and Navigation bar to use light theme.

Bug: 109947100
Test: Manual
Change-Id: I366f4d6a17eb6fccc9ca18ee638a82e5566b136f
---
 res/values/colors.xml                     | 15 ++++----
 res/values/dimens.xml                     |  3 +-
 res/values/styles.xml                     | 46 +++++++++++++----------
 res/xml/activity_blocked_numbers.xml      |  4 +-
 res/xml/add_blocked_number_dialog.xml     |  3 +-
 res/xml/blocking_suppressed_butterbar.xml |  6 +--
 res/xml/layout_blocked_number.xml         |  2 +-
 7 files changed, 44 insertions(+), 35 deletions(-)

diff --git a/res/values/colors.xml b/res/values/colors.xml
index 8ff0b8e4..0c7c175d 100644
--- a/res/values/colors.xml
+++ b/res/values/colors.xml
@@ -15,17 +15,16 @@
   -->
 
 <resources>
+    <!--follows the dialer theme color-->
     <color name="theme_color">#2a56c6</color>
+    <color name="background_color">#fafafa</color>
 
-    <color name="dialer_settings_actionbar_text_color">#ffffff</color>
-    <color name="dialer_settings_actionbar_background_color">@color/theme_color</color>
-    <color name="dialer_settings_color_dark">#1c3aa9</color>
-
-    <color name="blocked_numbers_divider_color">#e5e5e5</color>
-    <color name="blocked_numbers_butter_bar_color">#f5f5f5</color>
-    <color name="blocked_numbers_title_text_color">#ba000000</color>
-    <color name="blocked_numbers_secondary_text_color">#89000000</color>
+    <!--Use the one consistence with dialer-->
+    <color name="blocked_numbers_divider_color">#d8d8d8</color>
+    <color name="blocked_numbers_primary_text_color">#202124</color>
+    <color name="blocked_numbers_secondary_text_color">#5f6368</color>
 
+    <!--follows dialer color-->
     <color name="notification_action_answer">#097138</color>
     <color name="notification_action_decline">#A52714</color>
 </resources>
diff --git a/res/values/dimens.xml b/res/values/dimens.xml
index 7b085ad8..9986cb99 100644
--- a/res/values/dimens.xml
+++ b/res/values/dimens.xml
@@ -27,7 +27,8 @@
     <dimen name="blocked_numbers_dialog_padding">24dp</dimen>
     <dimen name="blocked_numbers_delete_icon_padding">12dp</dimen>
     <dimen name="blocked_numbers_progress_bar_padding">100dp</dimen>
-    <dimen name="blocked_numbers_font_size">16sp</dimen>
+    <dimen name="blocked_numbers_head1_font_size">18sp</dimen>
+    <dimen name="blocked_numbers_primary2_font_size">16sp</dimen>
     <dimen name="blocked_numbers_secondary_font_size">14sp</dimen>
     <dimen name="blocked_numbers_line_spacing">8sp</dimen>
     <dimen name="blocked_numbers_secondary_line_spacing">6sp</dimen>
diff --git a/res/values/styles.xml b/res/values/styles.xml
index 32167193..2e8e6242 100644
--- a/res/values/styles.xml
+++ b/res/values/styles.xml
@@ -26,58 +26,66 @@
     </style>
 
     <style name="Theme.Telecom.DialerSettings" parent="@android:style/Theme.Material.Light">
+        <item name="android:tint">@color/blocked_numbers_secondary_text_color</item>
         <item name="android:actionBarStyle">@style/TelecomDialerSettingsActionBarStyle</item>
         <item name="android:actionOverflowButtonStyle">@style/TelecomDialerSettingsActionOverflowButtonStyle</item>
-        <item name="android:colorPrimaryDark">@color/dialer_settings_color_dark</item>
-        <item name="android:homeAsUpIndicator">@drawable/ic_back_arrow</item>
+        <item name="android:colorPrimaryDark">@color/background_color</item>
+        <item name="android:windowLightStatusBar">true</item>
+        <item name="android:navigationBarColor">@color/background_color</item>
+        <item name="android:navigationBarDividerColor">@color/blocked_numbers_divider_color</item>
+        <item name="android:windowLightNavigationBar">true</item>
         <item name="android:windowContentOverlay">@null</item>
     </style>
 
     <style name="Theme.Telecom.BlockedNumbers" parent="@android:style/Theme.Material.Light">
+        <item name="android:tint">@color/blocked_numbers_secondary_text_color</item>
         <item name="android:actionBarStyle">@style/TelecomDialerSettingsActionBarStyle</item>
-        <item name="android:colorPrimaryDark">@color/dialer_settings_color_dark</item>
-        <item name="android:homeAsUpIndicator">@drawable/ic_back_arrow</item>
+        <item name="android:colorPrimaryDark">@color/background_color</item>
+        <item name="android:windowLightStatusBar">true</item>
+        <item name="android:navigationBarColor">@color/background_color</item>
+        <item name="android:navigationBarDividerColor">@color/blocked_numbers_divider_color</item>
+        <item name="android:windowLightNavigationBar">true</item>
         <item name="android:windowContentOverlay">@null</item>
         <item name="android:colorAccent">@color/theme_color</item>
         <item name="android:listDivider">@null</item>
     </style>
 
     <style name="TelecomDialerSettingsActionBarStyle" parent="android:Widget.Material.ActionBar">
-        <item name="android:background">@color/dialer_settings_actionbar_background_color</item>
-        <item name="android:titleTextStyle">@style/TelecomDialerSettingsActionBarTitleText</item>
+        <item name="android:background">@color/background_color</item>
+        <item name="android:titleTextStyle">@style/BlockedNumbersTextHead1</item>
         <item name="android:elevation">@dimen/dialer_settings_actionbar_elevation</item>
         <!-- Empty icon -->
         <item name="android:icon">@android:color/transparent</item>
     </style>
 
-    <style name="TelecomDialerSettingsActionBarTitleText"
-            parent="@android:style/TextAppearance.Material.Widget.ActionBar.Title">
-        <item name="android:textColor">@color/dialer_settings_actionbar_text_color</item>
-    </style>
-
     <style name="TelecomDialerSettingsActionOverflowButtonStyle"
             parent="@android:style/Widget.Material.Light.ActionButton.Overflow">
         <item name="android:src">@drawable/ic_more_vert_24dp</item>
     </style>
 
-    <style name="BlockedNumbersButton">
+    <style name="BlockedNumbersButton" parent="BlockedNumbersTextPrimary2">
         <item name="android:textColor">@color/theme_color</item>
-        <item name="android:textSize">@dimen/blocked_numbers_font_size</item>
         <item name="android:textAllCaps">true</item>
     </style>
 
-    <style name="BlockedNumbersText">
-        <item name="android:textSize">@dimen/blocked_numbers_font_size</item>
-        <item name="android:lineSpacingExtra">@dimen/blocked_numbers_line_spacing</item>
+    <style name="BlockedNumbersTextHead1"
+           parent="@android:style/TextAppearance.Material.Widget.ActionBar.Title">
+        <item name="android:textColor">@color/blocked_numbers_primary_text_color</item>
+        <item name="android:textSize">@dimen/blocked_numbers_head1_font_size</item>
+        <item name="android:fontFamily">sans-serif-regular</item>
     </style>
 
-    <style name="BlockedNumbersTitleText"  parent="BlockedNumbersText">
-        <item name="android:textColor">@color/blocked_numbers_title_text_color</item>
+    <style name="BlockedNumbersTextPrimary2">
+        <item name="android:textColor">@color/blocked_numbers_primary_text_color</item>
+        <item name="android:textSize">@dimen/blocked_numbers_primary2_font_size</item>
+        <item name="android:fontFamily">sans-serif-regular</item>
+        <item name="android:lineSpacingExtra">@dimen/blocked_numbers_line_spacing</item>
     </style>
 
-    <style name="BlockedNumbersSecondaryText">
+    <style name="BlockedNumbersTextSecondary">
         <item name="android:textColor">@color/blocked_numbers_secondary_text_color</item>
         <item name="android:textSize">@dimen/blocked_numbers_secondary_font_size</item>
+        <item name="android:fontFamily">sans-serif-regular</item>
         <item name="android:lineSpacingExtra">@dimen/blocked_numbers_secondary_line_spacing</item>
     </style>
 </resources>
diff --git a/res/xml/activity_blocked_numbers.xml b/res/xml/activity_blocked_numbers.xml
index 15e18590..e1373134 100644
--- a/res/xml/activity_blocked_numbers.xml
+++ b/res/xml/activity_blocked_numbers.xml
@@ -43,7 +43,7 @@
                     android:paddingTop="@dimen/blocked_numbers_large_padding"
                     android:paddingLeft="@dimen/blocked_numbers_large_padding"
                     android:paddingRight="@dimen/blocked_numbers_large_padding"
-                    style="@style/BlockedNumbersTitleText"
+                    style="@style/BlockedNumbersTextPrimary2"
                     android:visibility="gone" />
 
             <LinearLayout
@@ -70,7 +70,7 @@
                             android:layout_height="wrap_content"
                             android:text="@string/blocked_numbers_msg"
                             android:paddingBottom="@dimen/blocked_numbers_extra_large_padding"
-                            style="@style/BlockedNumbersTitleText" />
+                            style="@style/BlockedNumbersTextPrimary2" />
 
                     <TextView
                             android:id="@+id/add_blocked"
diff --git a/res/xml/add_blocked_number_dialog.xml b/res/xml/add_blocked_number_dialog.xml
index ed7d507b..d0942ecd 100644
--- a/res/xml/add_blocked_number_dialog.xml
+++ b/res/xml/add_blocked_number_dialog.xml
@@ -20,13 +20,14 @@
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:orientation="vertical"
+        android:gravity="center"
         android:padding="@dimen/blocked_numbers_dialog_padding">
     <TextView
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:text="@string/add_blocked_dialog_body"
             android:paddingBottom="@dimen/blocked_numbers_large_padding"
-            style="@style/BlockedNumbersTitleText" />
+            style="@style/BlockedNumbersTextPrimary2" />
     <EditText
             android:id="@+id/add_blocked_number"
             android:layout_width="match_parent"
diff --git a/res/xml/blocking_suppressed_butterbar.xml b/res/xml/blocking_suppressed_butterbar.xml
index d6a8472e..8b941b99 100644
--- a/res/xml/blocking_suppressed_butterbar.xml
+++ b/res/xml/blocking_suppressed_butterbar.xml
@@ -19,7 +19,7 @@
         xmlns:android="http://schemas.android.com/apk/res/android"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:background="@color/blocked_numbers_butter_bar_color">
+        android:background="@color/background_color">
 
     <ImageView
             android:id="@+id/icon"
@@ -39,7 +39,7 @@
             android:paddingTop="@dimen/blocked_numbers_large_padding"
             android:paddingRight="@dimen/blocked_numbers_large_padding"
             android:text="@string/blocked_numbers_butter_bar_title"
-            style="@style/BlockedNumbersTitleText" />
+            style="@style/BlockedNumbersTextPrimary2" />
 
     <TextView
             android:id="@+id/description"
@@ -51,7 +51,7 @@
             android:paddingBottom="@dimen/blocked_numbers_large_padding"
             android:paddingRight="@dimen/blocked_numbers_large_padding"
             android:text="@string/blocked_numbers_butter_bar_body"
-            style="@style/BlockedNumbersSecondaryText" />
+            style="@style/BlockedNumbersTextSecondary" />
 
     <TextView
             android:id="@+id/reenable_button"
diff --git a/res/xml/layout_blocked_number.xml b/res/xml/layout_blocked_number.xml
index fbd7de3d..720d71a5 100644
--- a/res/xml/layout_blocked_number.xml
+++ b/res/xml/layout_blocked_number.xml
@@ -20,7 +20,7 @@
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
     android:focusable="true"
-    style="@style/BlockedNumbersText"
+    style="@style/BlockedNumbersTextPrimary2"
     android:gravity="start">
 
     <TextView
-- 
2.17.1

